src/main/com/example010/radek/v2/Battle.java

```diff
@@ -0,0 +1,56 @@
+package main.com.example010.radek.v2;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Battle {
+
+    public void battle(int rounds) throws NoSuchAlgorithmException {
+
+        List<Soldier> soldiers = new ArrayList<>();
+        soldiers.add(new Fighter("Maximus Decimus Meridius", 10));
+        soldiers.add(new Universal("Commodus", 10));
+        for (int i = 0; i < rounds; i++) {
+            SecureRandom random = SecureRandom.getInstanceStrong();
+            int attacker = random.nextInt(soldiers.size());
+            for (int l = 0; l < soldiers.size(); l++) {
+                if (l != attacker) {
+                    soldiers.get(attacker).attack(soldiers.get(l));
+                }
+            }
+            printSoldierStats(soldiers);
+            soldiers = removeDeathFromList(soldiers);
+        }
+    }
+
+    public List<Soldier> removeDeathFromList(List<Soldier> soldiers) {

```

Comment:
1) lepsi nazev - removeDeathSoldiers(List..)
2) to co se deje v ifu uz nekoresponduje s nazvem metody
3) vracis list, opet nekoresponduje s nayvem ... do metodz, uz posilas list a tadz se s nim pracuje ...predava se ukazatel, tzn. ze se pracuj s tim listem, ktery do tedo metody vkladas,neni treba to vracet ... mrkni na example, kterz jsem ti commitnul pro predstavu...kdybys nechapal, tak se ptej

Reply:
1, přejmenováno 
2, rozumím, rozděleno
3, S tím předáváním listu jsem to nevěděl, ale člověk se pořád učí, je to tam

Reply:
přepiš for na foreach ... ze zacatku je to tezsi na pochopeni a preorientovani, ale v praxi z 99% pouzijes foreach, pokud neopracujes s nejakymi maticemi a nekde, kde to je fakt nutne